Naveen-gali/Yelp-App
# Carousel

### What does this PR do?

- Add A HomeScreen Carousel inspired by yelp app android design.
- Add Custom Button Component.


### Related Issues

- closes #40 

### How to test

- Clone the Repository.
- Switch to feat/carousel branch.
- Do yarn to install the dependencies.
- Do yarn run build-stage-ios for ios or yarn run build-stage-android for android.
- Now you can see the Home Screen with a carousel along with some text and a button.

### Screenshots

|Android ScreenShot           |  Ios ScreenShot|
|:-------------------------:|:-------------------------:
|![Screenshot_1681814023](https://user-images.githubusercontent.com/122969303/232753686-72dcc2c9-c6fa-4395-9128-64b9f16da96b.png) | ![Simulator Screen Shot - iPhone 14 - 2023-04-18 at 15 47 58](https://user-images.githubusercontent.com/122969303/232748609-e45129e0-39aa-4cbc-981e-84a7c33b87e1.png) |

### Checklist

### Additional Notes

- [Issue to use the dots pagination package](https://github.com/software-mansion/react-native-reanimated/issues/1460)

### Packages Added (if any)

| Package | Reason |
| ------- | ------ |
|react-native-reanimated         | Support package for react-native-reanimated-carousel|
| react-native-reanimated-carousel| Carousel Package |
| react-native-dots-pagination | Dots Kind of Pagination |
| react-native-gesture-handler | Gesture Handling Support Package |
